.github/workflows/swiftpm.yml

```diff
@@ -0,0 +1,38 @@
+name: SwiftPM Build CI
+
+on:
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+
+jobs:
+  swiftpm_macos_test:
+    name: Build with SwiftPM on macOS
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [macos-13, macos-14]
+        xcode-version: ["15.2"]
```

Comment:
Xcode version does not matter here on SwiftPM CI as we are using nightly Swift toolchain here.

Xcode 15.3 is not used because the upstream runner does not support it yet.

https://github.com/actions/runner-images/issues/9459